qemu/tests/watchdog.py

```diff
@@ -87,9 +87,17 @@ def _action_check(test, session, watchdog_action):
         response_timeout = int(params.get("response_timeout", '240'))
         error_context.context("Check whether or not watchdog action '%s' took"
                               " effect" % watchdog_action, logging.info)
+        if watchdog_action == "inject-nmi":
+            if not utils_misc.wait_for(lambda: "NMI received" in session.get_output(),
+                                       response_timeout, 0, 1):
+                test.fail("Guest didn't receive debug info with 'NMI received',"
+                          "so action '%s' didn't take effect." % watchdog_action)
+            msg = session.get_output().splitlines()[-8:]
+            logging.info("Guest received debug info: %s" % msg)
```

Comment:
Does log the `msg` here is necessary ?

Reply:
Yes, I think so. Because the criterion of the case pass is to check if the guest receive the nmi info.